### PR TITLE
Serialization: Remove old version checks

### PR DESCRIPTION
**Pull request by spinscale**
As the 2.x release does not need to be backwards compatible in terms of
serialization, we can remove a fair share of the serialization checks.

TODO: Unsure about the `MetaData` and the `Priority` version check removal. Just want to make sure, we dont persist any of those (snapshot/restore maybe?) - maybe someone can clarify.

### CODE REVIEW

File: src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java [hunk 1]
```diff
@@ -152,11 +151,6 @@ public void readFrom(StreamInput in) throws IOException {
         percolateCount = in.readVLong();
         percolateTimeInMillis = in.readVLong();
         current = in.readVLong();
-        if (in.getVersion().before(Version.V_1_1_0)) {
-            in.readVLong();
-        } else {
-            in.readLong();
```
Comment (dakrone) [inline src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java]: This needs to stay, right?

File: src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java [hunk 2]
```diff
@@ -165,11 +159,6 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(percolateCount);
         out.writeVLong(percolateTimeInMillis);
         out.writeVLong(current);
-        if (out.getVersion().before(Version.V_1_1_0)) {
-            out.writeVLong(0);
-        } else {
-            out.writeLong(-1);
```
Comment (dakrone) [inline src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java]: Same here, needs to stay?

**Comment by s1monw** [comment]
@spinscale even if we do persist them I have no idea how we would know that these are written by an old version. S/R uses XContent to persist stuff so I think it's all fine

**Comment by s1monw** [comment]
LGTM

File: src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java (hunk 1 above)
Comment (dakrone) [inline src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java]: Oh nevermind :)

File: src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java (hunk 2 above)
Comment (dakrone) [inline src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java]: Same here, nevermind again :)

File: src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java (hunk 1 above)
Comment (spinscale) [inline src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java]: not sure I follow you, why do we need reading and not using an always static value? Cant we just ditch it?

**Comment by dakrone** [comment]
LGTM too

File: src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java (hunk 1 above)
Comment (dakrone) [inline src/main/java/org/elasticsearch/index/percolator/stats/PercolateStats.java]: Yeah, sorry, it can be ditched, I read it incorrectly

**Pull request merged**